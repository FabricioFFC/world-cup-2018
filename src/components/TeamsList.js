import React, { Component } from 'react';
import { View, FlatList, ActivityIndicator } from 'react-native';

import TeamItem from './TeamItem';
import teamsApi from '../api/TeamsApi';

class TeamsList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      countries: [],
      loading: true,
    };
  }

  async componentWillMount() {
    const countries = await teamsApi.getAll();
    this.setState({ countries, loading: false });
  }

  render() {
    if (this.state.loading) {
      return (
        <View style={{ flex: 1, padding: 20 }}>
          <ActivityIndicator size="large" color="#0000ff" />
        </View>
      );
    }

    return (
      <FlatList
        data={this.state.countries}
        numColumns={2}
        keyExtractor={item => item.code}
        renderItem={({ item }) => <TeamItem item={item} />}
      />
    );
  }
}

export default TeamsList;
