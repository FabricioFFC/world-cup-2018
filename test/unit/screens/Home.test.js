import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import HomeScreen from '../../../src/screens/Home';
import TeamsList from '../../../src/components/TeamsList';


describe('HomeScreen', () => {
  describe('render', () => {
    it('show TeamsList component', () => {
      // given
      const shallowRenderer = new ShallowRenderer();
      // when
      shallowRenderer.render(<HomeScreen />);
      const result = shallowRenderer.getRenderOutput();
      // then
      expect(result.props.children).toEqual(<TeamsList />);
      expect(result).toMatchSnapshot();
    });
  });
});
