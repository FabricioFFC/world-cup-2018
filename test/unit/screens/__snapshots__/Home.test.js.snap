// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen render show TeamsList component 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#f4f5f9",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "paddingBottom": 25,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 25,
    }
  }
>
  <TeamsList />
</Component>
`;
